core/aws.py

```diff
@@ -20,18 +23,53 @@
     _S3_CLIENT = client(
         "s3",
         region_name=Config.AWS_REGION,
-        endpoint_url=Config.AWS_ENDPOINT_OVERRIDE,
     )
 
 
-def upload_file(file: FileIO, bucket: str, object_name: str):
+def upload_file(file: IO, bucket: str, object_name: str):
     """Uploads a file to an S3 bucket.
 
     :param file: a readable file-like object
     :param bucket: bucket to upload to
     :param object_name: S3 object name
     """
+    file.seek(0)
     try:
         _S3_CLIENT.upload_fileobj(file, bucket, object_name)
     except (ClientError, EndpointConnectionError) as bucket_error:
         current_app.logger.error(bucket_error)
+
+
+def get_file(bucket: str, object_name: str) -> BytesIO:
+    """Retrieves a file from an S3 bucket.
+
+    :param bucket: bucket to retrieve from
+    :param object_name: S3 object name
+    :return: retrieved file as a BytesIO
+    """
+    try:
+        file_like = BytesIO()
+        _S3_CLIENT.download_fileobj(bucket, object_name, file_like)
+        return file_like
```

Comment:
Had a go using this function and found it returned an iterable file object with the pointer set at the end of the file, so I needed to use .seek(0) to read from the start of the file. You could consider putting this into the function so that it returns a file ready to read :)

Reply:
yeah good shout